src/main/java/ch/uzh/ifi/hase/soprafs22/entity/Chat.java

```diff
@@ -46,17 +45,30 @@ public List<Message> getMessages(int fromIndex, int toIndex) {
     // now, fromIndex must be < size, and toIndex too.
     return messages.subList(fromIndex, toIndex);
   }
+    /**
+     * Gets one Message from the chat
+     * @param index: the index of the message
+     * @return Message: the Message on position index
+     */
+    public Message getMessages(int index){
+        if (messages.isEmpty()){
+            return null;
+        }
+        return messages.get(index);
+    }
+
+    /**
+     * Gets all the messages
+     * @return Message: List of all messages
+     */
+    public List<Message> getMessages() {
+        // return all
+        return messages;
+    }
 
   public void pushMessage(Message message) {
     // add at head, makes calculation easier
     messages.add(0, message);
   }
 
-  public Set<Long> getUserIds() {
-    return userIds;
-  }
-
-  public void setUserIds(Set<Long> userIds) {
-    this.userIds = userIds;
-  }
```

Comment:
I think this is needed when we create a new Chat? We did you remove it? (Must)

Reply:
Match has a User pair. becuase Match to chat is 1:1, we can get the users (and their id) from the match

Reply:
Fair point. (we might have to adapt match to not hold the users themselves but only the userId's (errors..))